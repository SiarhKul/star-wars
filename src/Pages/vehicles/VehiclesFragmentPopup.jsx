import React from "react";
import { useSelector } from "react-redux";
import { PopupListItem } from "../../components/popup/PopupListItem";

export const VehiclesFragmentPopup = () => {
	const {
		name,
		model,
		momanufacturerdel,
		cost_in_credits,
		length,
		max_atmosphering_speed,
		crew,
		passengers,
		cargo_capacity,
		consumables,
		vehicle_class,
		films,
		pilots,
	} = useSelector(state => state.app.clickedCard);

	return (
		<>
			<div className="name-wrapper">
				<span className="popup__name">{name}</span>
			</div>
			<div className="description-wrapper">
				<div className="description-person">
					<div className="description-appearance">
						<p className="description__heading">Appearance</p>
						<ul className="description-appearance__item">
							<li>Model:{model}</li>
							<li>Manufacturer:{momanufacturerdel}</li>
							<li>Cost in credits:{cost_in_credits}</li>
							<li>Length:{length}</li>
							<li>Max atmosphering speed:{max_atmosphering_speed}</li>
							<li>Crew:{crew}</li>
							<li>Passengers:{passengers}</li>
							<li>Cargo capacity:{cargo_capacity}</li>
							<li>Consumables:{consumables}</li>
							<li>Vehicle_class:{vehicle_class}</li>
						</ul>
					</div>
				</div>

				<div className="description-resources">
					<div className="resources">
						<div className="resources-heading">Films:</div>
						<ul className="resources-list">
							<PopupListItem urls={films} name="title" path="films" />
						</ul>
					</div>

					<div className="resources">
						<div className="resources-heading">Pilots:</div>
						<ul className="resources-list">
							<PopupListItem urls={pilots} name="name" path="" />
						</ul>
					</div>
				</div>
			</div>
		</>
	);
};
