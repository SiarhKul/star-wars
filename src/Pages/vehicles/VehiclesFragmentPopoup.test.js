import { shallow } from "enzyme";
import React from "react";
import { VehiclesFragmentPopup } from "./VehiclesFragmentPopup";

jest.mock("history", () => {
	return {
		createBrowserHistory: jest.fn(() => {
			return {
				location: {
					state: {
						state: {
							name: "X-34 landspeeder",
							model: "X-34 landspeeder",
							manufacturer: "SoroSuub Corporation",
							cost_in_credits: "10550",
							length: "3.4 ",
							max_atmosphering_speed: "250",
							crew: "1",
							passengers: "1",
							cargo_capacity: "5",
							consumables: "unknown",
							vehicle_class: "repulsorcraft",
							pilots: [],
							films: ["https://swapi.dev/api/films/1/"],
						},
					},
				},
			};
		}),
	};
});

describe("Test <VehiclesFragmentPopup/>", () => {
	const component = shallow(<VehiclesFragmentPopup />);

	it("render snapshot <VehicleFragmentPopup/> ", () => {
		expect(component).toMatchSnapshot();
	});
	it("should  <VehiclesFragmentPopup/>  has required fields", () => {});
	const requiredFields = [
		"Appearance",
		"Model:",
		"Manufacturer:",
		"Cost in credits:",
		"Length:",
		"Max atmosphering speed:",
		"Crew:",
		"Passengers:",
		"Cargo capacity:",
		"Consumables:",
		"Vehicle_class:",
		"Films:",
		"Pilots:",
	];
	expect(component.containsAllMatchingElements(requiredFields)).toBeTruthy();
});
