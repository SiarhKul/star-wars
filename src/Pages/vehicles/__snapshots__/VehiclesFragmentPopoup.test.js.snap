// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <VehiclesFragmentPopup/> render snapshot <VehicleFragmentPopup/>  1`] = `
<Fragment>
  <div
    className="name-wrapper"
  >
    <span
      className="popup__name"
    />
  </div>
  <div
    className="description-wrapper"
  >
    <div
      className="description-person"
    >
      <div
        className="description-appearance"
      >
        <p
          className="description__heading"
        >
          Appearance
        </p>
        <ul
          className="description-appearance__item"
        >
          <li>
            Model:
          </li>
          <li>
            Manufacturer:
          </li>
          <li>
            Cost in credits:
          </li>
          <li>
            Length:
          </li>
          <li>
            Max atmosphering speed:
          </li>
          <li>
            Crew:
          </li>
          <li>
            Passengers:
          </li>
          <li>
            Cargo capacity:
          </li>
          <li>
            Consumables:
          </li>
          <li>
            Vehicle_class:
          </li>
        </ul>
      </div>
    </div>
    <div
      className="description-resources"
    >
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Films:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="title"
            path="films"
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Pilots:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="people"
          />
        </ul>
      </div>
    </div>
  </div>
</Fragment>
`;
